sonar-plugin-api/src/main/java/org/sonar/api/scan/filesystem/PathResolver.java

```diff
@@ -73,19 +67,32 @@ public RelativePath relativePath(Collection<File> dirs, File file) {
     return null;
   }
 
+  /**
+   * Similar to {@link Path#relativize(Path)} except that:
+   *   <ul>
+   *   <li>null is returned if file is not a child of dir
+   *   <li>the resulting path is converted to use Unix separators
+   *   </ul> 
+   * @since 6.0
+   */
   @CheckForNull
-  public String relativePath(File dir, File file) {
-    List<String> stack = new ArrayList<>();
-    String dirPath = PathUtils.canonicalPath(dir);
-    File cursor = file;
-    while (cursor != null) {
-      if (dirPath.equals(PathUtils.canonicalPath(cursor))) {
-        return Joiner.on("/").join(stack);
-      }
-      stack.add(0, cursor.getName());
-      cursor = cursor.getParentFile();
+  public String relativePath(Path dir, Path file) {
+    Path baseDir = dir.normalize();
+    Path path = file.normalize();
+    if (!path.startsWith(baseDir)) {
+      return null;
     }
-    return null;
+    try {
+      Path relativized = baseDir.relativize(path);
+      return FilenameUtils.separatorsToUnix(relativized.toString());
+    } catch (IllegalArgumentException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)